posthog/api/test/test_organization_feature_flag.py

```diff
@@ -44,3 +47,87 @@ def test_get_feature_flag_unauthorized(self):
         response = self.client.get(url)
 
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class TestOrganizationFeatureFlagCopy(APIBaseTest, QueryMatchingTest):
+    def setUp(self):
+        self.team_1 = self.team
+        self.team_2 = Team.objects.create(organization=self.organization)
+
+        self.feature_flag_key = "copied-flag-key"
+        self.feature_flag_to_copy = FeatureFlag.objects.create(
+            team=self.team_1,
+            created_by=self.user,
+            key=self.feature_flag_key,
+            filters={"groups": [{"rollout_percentage": 65}]},
+        )
+
+        super().setUp()
+
+    @snapshot_postgres_queries
+    def test_copy_feature_flag_success(self):
+        url = f"/api/organizations/{self.organization.id}/feature_flags/copy_flags"
+        data = {
+            "feature_flag_key": self.feature_flag_to_copy.key,
+            "from_project": self.feature_flag_to_copy.team_id,
+            "target_project_ids": [self.team_2.id],
+        }
+        response = self.client.post(url, data)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn("success", response.json())
+        self.assertIn("failed", response.json())
+
+        # copied flag is present in the response
+        self.assertEqual(self.feature_flag_key, response.json()["success"][0]["key"])
+
+        # copied flag is present in the DB
+        copied_flag = FeatureFlag.objects.filter(team=self.team_2, key=self.feature_flag_key).exists()
+        self.assertTrue(copied_flag)
```

Comment:
also test  all the right fields are in the flag :) (specifically the two I mentioned above we missed in the copy), and also check all other fields are not present.

As a bonus, would be ideal if this test fails whenever a new field is added to the feature flag model, so we're aware to check here & confirm if something needs to be added to the model. You can do this by asserting the full serialised response is what we'd expect.

Reply:
I'm not sure how to do this correctly.

- I see some additional fields in the model besides those you suggested to add (e.g. performed_rollback, analytics_dashboard)
- Some fields, I suppose, are added automatically (e.g. created_by), so we don't want to copy them explicitly.
- Some fields seem optional, as I see them in the model but not in the response or the Django admin (e.g. analytics dashboard)?

Anyway, I added an assertion for those fields we are copying explicitly. Can you suggest a change to address what's missing here? :)

Reply:
created_by is added because of the `context` parameter, so worth checking as well, it should be same as the current user `self.user`